sale_order_line_date_next_reception/models/sale_order_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Akretion (http://www.akretion.com).
+# @author Mathieu DELVA <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    date_next_reception = fields.Date(compute="_compute_date_next_reception", compute_sudo=True)
+
+    def _compute_date_next_reception(self):
+        for line in self:
+            line.date_next_reception = False
+            if not(line.product_id.qty_available) and line.state not in ['done', 'cancel']:
```

Comment:
You should test "line.product_id.qty_available < 0" as the qty available can be negatif